signer/threshold_validator.go

```diff
@@ -377,31 +372,30 @@ func (pv *ThresholdValidator) waitForPeerNonces(
 	}
 	thresholdPeersMutex.Unlock()
 }
-
 func (pv *ThresholdValidator) waitForPeerSetNoncesAndSign(
 	chainID string,
-	ourID int,
 	peer Cosigner,
 	hrst HRSTKey,
-	encryptedNoncesThresholdMap map[Cosigner][]CosignerNonce,
+	noncesMap map[Cosigner][]CosignerNonce,
 	signBytes []byte,
 	shareSignatures *[][]byte,
 	shareSignaturesMutex *sync.Mutex,
-	noncePublic *[]byte,
 	wg *sync.WaitGroup,
 ) {
 	peerStartTime := time.Now()
 	defer wg.Done()
 	peerNonces := make([]CosignerNonce, 0, pv.threshold-1)
 
-	for _, encryptedSecrets := range encryptedNoncesThresholdMap {
-		for _, ephemeralSecretPart := range encryptedSecrets {
+	peerID := peer.GetID()
+
+	for _, nonces := range noncesMap {
+		for _, nonce := range nonces {
 			// if share is intended for peer, check to make sure source peer is included in threshold
-			if ephemeralSecretPart.DestinationID == peer.GetID() {
-				for thresholdPeer := range encryptedNoncesThresholdMap {
-					if thresholdPeer.GetID() == ephemeralSecretPart.SourceID {
+			if nonce.DestinationID == peerID {
```

Comment:
Inverting this condition (`if nonce.DestinationID != peerID { continue }`) would reduce a level of indentation here, which would help readability since we reach five levels deep.